http_2.rflx

```diff
@@ -0,0 +1,230 @@
+				then Exclusive_Flag
+					if Frame_Type = PRIORITY
+				then Error_Code
+					if Frame_Type = RST_STREAM
+				then Settings_Identifier
+					if Frame_Type = SETTINGS
+				then Data
+					with Size => 64
+					if Frame_Type = PING and Stream_Identifier = 0 -- RFC: If a PING frame is received with a stream identifier field value other than 0x0, the recipient MUST respond with a connection error (Section 5.4.1) of type PROTOCOL_ERROR.
+				then Reserved_Bit_2
+					if (Frame_Type = GOAWAY and Stream_Identifier = 0) or
+						Frame_Type = WINDOW_UPDATE or
+						(Frame_Type = PUSH_PROMISE and Flag_Padded = False);
+			Settings_Identifier : Settings_Parameter
+				then Settings_Value_Enable_Push
+					if Settings_Identifier = SETTINGS_ENABLE_PUSH
+				then Settings_Value_Initial_Window_Size
+					if Settings_Identifier = SETTINGS_INITIAL_WINDOW_SIZE
+				then Settings_Value_Max_Frame_Size
+					if Settings_Identifier = SETTINGS_MAX_FRAME_SIZE
+				then Settings_Value_Generic
+					if (Settings_Identifier = SETTINGS_HEADER_TABLE_SIZE or
+						Settings_Identifier = SETTINGS_MAX_CONCURRENT_STREAMS or
+						Settings_Identifier = SETTINGS_MAX_HEADER_LIST_SIZE);
+			Settings_Value_Enable_Push : Settings_Value_Enable_Push
+				then null;
+			Settings_Value_Initial_Window_Size : Settings_Value_Initial_Window_Size
+				then null;
+			Settings_Value_Max_Frame_Size : Settings_Value_Max_Frame_Size
+				then null;
+			Settings_Value_Generic : Settings_Value_Generic
+				then null;
+			Pad_Length : Pad_Length
+				then Data
+					with Size => (Payload_Length - (Pad_Length + 1)) * 8 	-- 1 -> Pad_Length'Length
+					if Frame_Type = DATA and Payload_Length >= Pad_Length + 1
+				then Exclusive_Flag
+					if Frame_Type = HEADERS and Flag_Priority = True
+				then Header_Block_Fragment
+					with Size => Payload_Length * 8
+					if Frame_Type = HEADERS and Flag_Priority = False
+				then Reserved_Bit_2
+					if Frame_Type = PUSH_PROMISE;
+			Reserved_Bit_2 : Zero_Bit
+				then Promised_Stream_ID
+					if Frame_Type = PUSH_PROMISE
+				then Last_Stream_ID 
+					if Frame_Type = GOAWAY
+				then Window_Size_Increment
+					if Frame_Type = WINDOW_UPDATE;
+			Promised_Stream_ID : Stream_ID
+				then Header_Block_Fragment
+					with Size => (Payload_Length - (Pad_Length + 5)) * 8
+					if Flag_Padded = True and Payload_Length >= Pad_Length + 5
+				then Header_Block_Fragment
+					with Size => (Payload_Length - 4) * 8
+					if Flag_Padded = False and Payload_Length >= 4;
+			Last_Stream_ID : Stream_ID
+				then Error_Code;
+			Exclusive_Flag : Boolean
+				then Stream_Dependency;
+			Stream_Dependency : Stream_ID
+				then Weight;
+			Weight : Weight
+				then Header_Block_Fragment
+					with Size => (Payload_Length - (Pad_Length + 6)) * 8
+					if Frame_Type = HEADERS and Flag_Padded = True and Payload_Length >= Pad_Length + 6
+				then Header_Block_Fragment
+					with Size => (Payload_Length - 5) * 8				 -- Sub prev field lengths of frame type
```

Comment:
Instead of using a constant (`5` in this case), the length of the previous fields should be represented by an expression (e.g.,  `B'Last - A'First + 1`).

Reply:
Changed in d40f173